pkg/kubelet/certificate/bootstrap/bootstrap.go

```diff
@@ -207,3 +217,30 @@ func verifyKeyData(data []byte) bool {
 	_, err := certutil.ParsePrivateKeyPEM(data)
 	return err == nil
 }
+
+func waitForServer(cfg restclient.Config, deadline time.Duration) error {
+	cfg.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
+	cfg.Timeout = 1 * time.Second
+	cli, err := restclient.UnversionedRESTClientFor(&cfg)
+	if err != nil {
+		return fmt.Errorf("couldn't create client: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.TODO(), deadline)
+	defer cancel()
+
+	var connected bool
+	wait.JitterUntil(func() {
+		if err := cli.Get().AbsPath("/healthz").Do().Error(); err != nil {
```

Comment:
do non-200 errors trigger this error path, or just network errors?

Reply:
non-200 errors do as well. We'd be happy with just waiting until we don't see network errors.

Reply:
Changed to network errors.